docs/DeveloperGuide.md

```diff
@@ -5,9 +5,262 @@
+
+### Update Command
+The following is a class diagram of the UpdateCommand and its relevant dependencies<br>
+![UpdateCommandClass](uml-diagrams/UpdateCommandClass.png)
+
+The `UpdateCommand` class implements the `Command` interface and is responsible for updating existing items in the 
+`Inventory`. A UpdateCommand instance is created by the `parseUpdateCommand` method called by Parser, which ensures 
+that the provided parameters (name, quantity, price) are valid. While it is optional to include quantity and price, one
+of the aforementioned parameters must be provided.
+
+#### Dependencies
+- `Item`: For getting the quantity and price of the to be updated item
+- `Inventory`: For updating the item in the inventory
+- `Ui`: To notify the user about the successful execution of `UpdateCommand`
+
+The following sequence diagram shows the execution of a UpdateCommand<br>
+![UpdateCommandSequence](uml-diagrams/UpdateCommandSequence.png)
+
+1. The `SuperTracker` class calls the `execute` method of `UpdateCommand`
+2. The item object of the item to be updated is obtained from `inventory`
+3. There is an optional check for newQuantity being -1 (an invalid value that indicates that quantity should not be 
+updated). If the condition holds true it retrieves the item's previous quantity.
+4. There is another optional check for newPrice being -1 (an invalid value that indicates that price should not be
+   updated). If the condition holds true it retrieves the item's previous price.
+5. A new `Item` object with the given parameters (name, new quantity, new price, old expiry date) is created and returned to `UpdateCommand`
+6. The `put` method of the `Inventory` class is called to update the item in the inventory
+7. The `UpdateCommandSuccess` method of the `Ui` class is called to notify that `UpdateCommand` has been successfully executed
+8. The `saveData` method of the `FileManager` class is called to save updated item onto the hard disk
+
+### DeleteCommand
+The following is a class diagram of the DeleteCommand and its relevant dependencies<br>
+![DeleteCommandClass](uml-diagrams/DeleteCommandClass.png)
+
+The `DeleteCommand` class implements the `Command` interface and is responsible for deleting existing items in the
+inventory. A DeleteCommand instance is created when calling the `parseDeleteCommand` method called in Parser class.
+This method parses the input and ensures that the command parameter (item name) exists in the inventory. The `execute()`
+method in the class will call the `delete` method from `Inventory` class to remove the item. It will then execute
+the `saveData` method from `FileManager` class to save changes to the inventory.
+
+#### Dependencies
+- `Inventory`: Checking and deleting item from inventory
+- `Ui`: To notify the user about the successful execution of `DeleteCommand`
+
+The following sequence diagram shows the execution of a DeleteCommand<br>
+![DeleteCommandSequence](uml-diagrams/DeleteCommandSequence.png)
```

Comment:
Activation bar for `Inventory` is not consistent. `contains()` is green while `delete()` is white.